validator/sawtooth_validator/gossip/signature_verifier.py

```diff
@@ -150,6 +178,19 @@ def handle(self, connection_id, message_content):
             self._seen_cache[obj.header_signature] = None
             return HandlerResult(status=HandlerStatus.PASS)
 
+        if tag == GossipMessage.CONSENSUS:
+            if obj.signature in self._seen_cache:
+                self._msg_dropped_count.inc()
+                return HandlerResult(status=HandlerStatus.DROP)
```

Comment:
I don't think we want to drop these duplicates yet. We're not gossiping, so duplicates should be low, and we don't have anything like a nonce in the consensus message, so it is realistic that messages will be received with identical signatures.

That being said, this probably means these messages are a denial of service vector since a malicious validator can spam the same message as fast as possible to slow down a peer. Dropping duplicates here and requiring nonces in duplicate consensus messages would mitigate this since a bad validator would need to create and sign a new message every time, which is more expensive than just resending the same message.